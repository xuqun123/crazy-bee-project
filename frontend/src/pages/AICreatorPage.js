import * as React from 'react'
import GenerateBar from '../components/GenerateBar'
import TopSlogan from '../components/TopSlogan'
import Container from '@mui/material/Container'
import Typography from '@mui/material/Typography'
import ArtworkDisplay from '../components/ArtworkDisplay'

function AICreatorPage() {
  return (
    <>
      <TopSlogan
        header="Create Unique Art with our AI Generator"
        summary="Enter your prompt below and start creating your own art!"
      />
      <ArtworkDisplay />
      <GenerateBar />
      <Container sx={{ py: 12 }} maxWidth="lg">
        <Typography gutterBottom variant="h5" component="h2">
          Frequently Asked Questions
        </Typography>
        <Typography gutterBottom>
          Can I use this for commercial use? Yes, all commercial use is allowed for the generated
          images.
        </Typography>
        <Typography gutterBottom>Can I use the generated images for NFT? Yes.</Typography>
        <Typography gutterBottom>
          Who owns the output? The images are considered public domain, that is, they have no owner.
        </Typography>
      </Container>
    </>
  )
}

export default AICreatorPage
