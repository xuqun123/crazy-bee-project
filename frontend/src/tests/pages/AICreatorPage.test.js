import React from 'react'
import { render, screen } from '@testing-library/react'
import AICreatorPage from '../../pages/AICreatorPage'
import { MemoryRouter } from 'react-router-dom'

describe('AICreatorPage', () => {
  afterEach(() => jest.clearAllMocks())

  describe('get users data via api', () => {
    it('render the AICreatorPage view properly with expected elements', async () => {
      render(<AICreatorPage />, { wrapper: MemoryRouter })

      const sloganHeader = screen.getByText('Create Unique Art with our AI Generator')
      expect(sloganHeader).toBeInTheDocument()

      const faqText = screen.getByText('Frequently Asked Questions')
      expect(faqText).toBeInTheDocument()
    })

    it('snapshot the AICreatorPage view with all expected elements', async () => {
      const view = render(<AICreatorPage />, { wrapper: MemoryRouter })
      expect(view).toMatchSnapshot()
    })
  })
})
