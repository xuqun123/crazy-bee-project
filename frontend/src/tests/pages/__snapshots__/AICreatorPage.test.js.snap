// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AICreatorPage get users data via api snapshot the AICreatorPage view with all expected elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-10m0cap"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
        >
          <h1
            class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
          >
            Create Unique Art with our AI Generator
          </h1>
          <p
            class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
          >
            Enter your prompt below and start creating your own art!
          </p>
          <div
            class="css-d1ev6n-MuiStack-root"
          />
        </div>
      </div>
      <form
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-9dmh1q-MuiPaper-root"
      >
        <div
          class="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
        >
          <input
            aria-label="Generate Artwork"
            class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
            placeholder="Input prompt..."
            type="text"
            value=""
          />
        </div>
        <button
          aria-label="generate"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="submit"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ArrowForwardIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-tj4w3q-MuiContainer-root"
      >
        <h2
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        >
          Frequently Asked Questions
        </h2>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          Can I use this for commercial use? Yes, all commercial use is allowed for the generated images.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          Can I use the generated images for NFT? Yes.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
        >
          Who owns the output? The images are considered public domain, that is, they have no owner.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-10m0cap"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
      >
        <h1
          class="MuiTypography-root MuiTypography-h2 MuiTypography-alignCenter MuiTypography-gutterBottom css-10759ka-MuiTypography-root"
        >
          Create Unique Art with our AI Generator
        </h1>
        <p
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter MuiTypography-paragraph css-17rfbpd-MuiTypography-root"
        >
          Enter your prompt below and start creating your own art!
        </p>
        <div
          class="css-d1ev6n-MuiStack-root"
        />
      </div>
    </div>
    <form
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-9dmh1q-MuiPaper-root"
    >
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary css-piqts5-MuiInputBase-root"
      >
        <input
          aria-label="Generate Artwork"
          class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
          placeholder="Input prompt..."
          type="text"
          value=""
        />
      </div>
      <button
        aria-label="generate"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1gtpq4r-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-tj4w3q-MuiContainer-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
      >
        Frequently Asked Questions
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
      >
        Can I use this for commercial use? Yes, all commercial use is allowed for the generated images.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
      >
        Can I use the generated images for NFT? Yes.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-d3wcwz-MuiTypography-root"
      >
        Who owns the output? The images are considered public domain, that is, they have no owner.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
